src/scan-imports.ts

```diff
@@ -77,45 +86,34 @@ function parseWebModuleSpecifier(specifier: string): null | string {
 }
 
 function getInstallTargetsForFile(filePath: string, code: string): InstallTarget[] {
+  const [imports] = parse(code) || [];
   const allImports: InstallTarget[] = [];
-  try {
-    const ast = parse(code, {plugins: ['dynamicImport'], sourceType: 'module'});
-    traverse(ast, {
-      ImportDeclaration(path) {
-        const webModuleSpecifier = parseWebModuleSpecifier(path.node.source.value);
-        if (webModuleSpecifier) {
-          allImports.push({
-            specifier: webModuleSpecifier,
-            all: false,
-            default: path.node.specifiers.some(s => s.type === 'ImportDefaultSpecifier'),
-            namespace: path.node.specifiers.some(s => s.type === 'ImportNamespaceSpecifier'),
-            named: path.node.specifiers
-              .map(s => s.type === 'ImportSpecifier' && s.imported.name)
-              .filter(Boolean),
-          });
-        }
-      },
-      Import(path) {
-        // Only match dynamic imports that are called as a function
-        if (path.parent.type !== 'CallExpression') {
-          return;
-        }
-        // Only match dynamic imports called with a single string argument
-        const [argNode] = path.parent.arguments;
-        if (argNode.type !== 'StringLiteral') {
-          return;
-        }
-        // Analyze that string argument as an import specifier
-        const webModuleSpecifier = parseWebModuleSpecifier(argNode.value);
-        if (webModuleSpecifier) {
-          allImports.push(createInstallTarget(webModuleSpecifier, true));
-        }
-      },
+
+  for (const {s: start, e: end, ss: sStart, se: sEnd, d: dynamicImportIndex} of imports) {
+    const webModuleSpecifier = parseWebModuleSpecifier(code.substring(start, end));
+    if (!webModuleSpecifier) {
+      continue;
+    }
+
+    const importStatement = code.substring(sStart, sEnd);
+    const dynamicImport = dynamicImportIndex > -1;
```

Comment:
nit: it might be nice to push all of this logic into a seperate function as well, so that you could just do something like:

```
allImports.push(parseImport(importStatement));
```